src/main/java/mate/academy/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.dao.OrderDao;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Order;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.OrderService;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(ShoppingCart shoppingCart) {
+        Order order = new Order();
+        List<Ticket> tickets = new ArrayList<>(shoppingCart.getTickets());
+        order.setTickets(tickets);
+        order.setUser(shoppingCart.getUser());
+        order.setOrderDate(LocalDateTime.now());
+        shoppingCartService.clearShoppingCart(shoppingCart);
+        orderDao.add(order);
+        shoppingCart.setTickets(tickets);
+        shoppingCartDao.update(shoppingCart);
```

Comment:
What is it for?

Reply:
shoppingCart.setTickets(tickets); - adds tickets
shopping Cart Dao.update(shopping Cart);  - updates the shopping cart that was passed to the method